### PR TITLE
feat: upload image show loading progress and close dialog

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### Issue

file upload dialog was not updating image correctly

[Link to Basecamp Todo](https://3.basecamp.com/3105655/projects)

### Solution

it now updates the image and even shows loading state and handles errors

### CODE REVIEW

File: apps/videos-admin/src/libs/SnackbarProvider/SnackbarProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use client'
```
Comment (edmonday) [inline apps/videos-admin/src/libs/SnackbarProvider/SnackbarProvider.tsx:1]: Is this file necessary? This seems like a wrapper for SnackbarProvider which I think we can just use as is?

**Review by edmonday — CHANGES_REQUESTED** [review]

File: apps/videos-admin/src/libs/SnackbarProvider/SnackbarProvider.tsx (hunk 1 above)
Comment (Kneesal) [reply apps/videos-admin/src/libs/SnackbarProvider/SnackbarProvider.tsx:1]: Yes see https://nextjs.org/docs/14/app/building-your-application/rendering/composition-patterns#using-third-party-packages-and-providers

Comment (edmonday) [reply apps/videos-admin/src/libs/SnackbarProvider/SnackbarProvider.tsx:1]: tyvm for the info

**Review by edmonday — APPROVED** [review]
